libnm-core/nm-setting-gsm.c

```diff
@@ -621,10 +614,10 @@ nm_setting_gsm_class_init (NMSettingGsmClass *klass)
 	/**
 	 * NMSettingGsm:number:
 	 *
-	 * Number to dial when establishing a PPP data session with the GSM-based
-	 * mobile broadband network.  Many modems do not require PPP for connections
-	 * to the mobile network and thus this property should be left blank, which
-	 * allows NetworkManager to select the appropriate settings automatically.
+	 * Legacy setting that used to help establishing PPP data sessions for
+	 * GSM-based modems.
+	 *
+	 * Deprecated: 1.16: user-provided values for this setting are no longer used..
```

Comment:
Double ".." at the end of sentence.

---

If you modify any of the gtk-doc settings, you need to regenerate `clients/common/settings-docs.h.in`. You'll also see that `make check` fails (if you build with gtk-doc). To fix it, `./configure` with `--enable-gtk-doc` and then run

```
   NM_TEST_REGENERATE=1 make -j 10 check
```

---

For other reasons, you also need to regenerate `clients/tests/test-client.check-on-disk/test_003.expected`. Doing that is a bit more tricky. See howto at [test-client.py](https://gitlab.freedesktop.org/NetworkManager/NetworkManager/blob/90e9695af541d0e13c5327466f1414b35a04c73d/clients/tests/test-client.py#L14). Optimally, the howto there is sufficiently clear. If you struggle understanding what you need to so, please ask and we see how the howto can be improved.

Reply:
For some reasons, this "Deprecated" line confuses `tools/check-settings-docs.sh`. I wonder whether the comment should be adjusted, or the script improved...

Reply:
I tried to follow gtk-doc style to write the deprecation comment, not sure if that's how you usually do it.

Reply:
Does this ring any bell? This is when I run `NM_TEST_REGENERATE=1 make check-local-clients-tests-test-client`

NM_TEST_CLIENT_NMCLI_PATH=clients/cli/nmcli \
"/usr/bin/python" \
./clients/tests/test-client.py -v &> "./clients/tests/test-client.log" && r=ok; \
cat "./clients/tests/test-client.log"; \
test "$r" == ok
test_001 (__main__.TestNmcli) ... ERROR
test_002 (__main__.TestNmcli) ... ERROR
test_003 (__main__.TestNmcli) ... ERROR
test_004 (__main__.TestNmcli) ... ERROR

======================================================================
ERROR: test_001 (__main__.TestNmcli)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./clients/tests/test-client.py", line 784, in f
    self._nm_test_post()
  File "./clients/tests/test-client.py", line 766, in _nm_test_post
    content_new = ''.join([r['content'] for r in results])
TypeError: sequence item 0: expected str instance, bytes found

======================================================================
ERROR: test_002 (__main__.TestNmcli)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./clients/tests/test-client.py", line 784, in f
    self._nm_test_post()
  File "./clients/tests/test-client.py", line 766, in _nm_test_post
    content_new = ''.join([r['content'] for r in results])
TypeError: sequence item 0: expected str instance, bytes found

======================================================================
ERROR: test_003 (__main__.TestNmcli)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./clients/tests/test-client.py", line 784, in f
    self._nm_test_post()
  File "./clients/tests/test-client.py", line 766, in _nm_test_post
    content_new = ''.join([r['content'] for r in results])
TypeError: sequence item 0: expected str instance, bytes found

======================================================================
ERROR: test_004 (__main__.TestNmcli)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./clients/tests/test-client.py", line 784, in f
    self._nm_test_post()
  File "./clients/tests/test-client.py", line 766, in _nm_test_post
    content_new = ''.join([r['content'] for r in results])
TypeError: sequence item 0: expected str instance, bytes found

----------------------------------------------------------------------
Ran 4 tests in 7.119s

Reply:
Wait, I know what that is. I'm using Arch Linux, where /usr/bin/python is really python3. After reconfiguring with PYTHON=/usr/bin/python2 explicitly, it worked.

Reply:
I believe I did the changes now

Reply:
that's a bit odd, because the test is supposed to work equally well with python2 and pyhon3. I am pretty sure we also test both versions (successfully).